src/parser/utils.rs

```diff
@@ -314,6 +317,42 @@ pub fn mk_label(types: Types, src_id: FileId, l: usize, r: usize) -> Label {
     }
 }
 
+/// Create a let/let pattern term. If the parsed let statement has a pattern we check if the term
+/// is also parsed as being recursive, we throw an error. This error is thrown because Nickel does
+/// not implement recursive let patterns, although it might in the future.
+///
+/// If no pattern was found, an, optionally recursive, let binding is constructed.
```

Comment:
```suggestion
/// Generate a `Let` or a `LetPattern` (depending on `pat` being empty or not) from a the parsing
/// of a let definition. This function fails if the definition has both a non-empty pattern and
/// is recursive (`pat != Destruct::Empty && rec`), because recursive let-patterns are currently
/// not supported.
```